### DISCUSSION TITLE
How to configure Next.js on azure?

### DISCUSSION THREAD

**Discussion by gabrielgrs** [vercel/next.js#28356] [Help]
I'm trying to upload an application to Azure and I'm having difficulties. Using a custom server, together with a next export, it works statically. But I need the SSR, I tried a few ways and I couldn't.

Would anyone know help me with this? Any tutorials or something?

Thank you!

**Comment by Alwinator** [comment]
This might not solve your problem, but I am using https://vercel.com for Next.JS Projects and it works perfectly.

**Reply by gabrielgrs** [reply]
Unfortunately not, for corporate decisions, I need it to be on azure :(

**Reply by Alwinator** [reply]
Maybe [this](https://stackoverflow.com/questions/64758494/deploy-stripe-payment-gateway-on-azure/64783120#64783120) helps.

**Reply by gabrielgrs** [reply]
Unfortunately not too, I already tryed this :(

**Reply by wwwdepot** [reply]
You can deploy Nodejs app directly in vscode, [see this guide](https://docs.microsoft.com/en-us/azure/app-service/quickstart-nodejs?pivots=platform-linux).

**Reply by alex-olivera** [reply]
@gabrielgrs any update for this? I'm trying too this

**Comment by pseudoramble** [comment]
I've done this where I work too. I don't have a particular tutorial that I have outside of [this guide](https://docs.microsoft.com/en-us/azure/app-service/quickstart-nodejs?pivots=development-environment-azure-portal&tabs=linux) which it seems like you noticed. A few notes that may help though:

- I used an Azure App Service running Linux. I attempted Windows at one point but was unable to get that working. It's been a long time since I set it up, so I don't recall the specific Windows issue I was having.
- One thing not clearly spelled out in that guide is that to execute the startup command from your `package.json` you'll need to add that into the App Service's Configuration. There's a way to specify the command in one of the tabs within that page, which I use to explicitly have it run the startup command from my package.json.
- Another thing that is explained but may not be obvious at first glance is that depending on how you upload your app, `node_modules` may have to be added to the App Service. [See here](https://docs.microsoft.com/en-us/azure/app-service/configure-language-nodejs?pivots=platform-linux). So make sure you keep that in mind during your setup.
- To debug it's very helpful to use the command line or SSH tools built in. There you can confirm that the package you upload is getting unpacked as expected, check that node_modules are there as expected, etc.

Hope this helps somebody out there.

**Comment by aaronpowell** [comment]
I know that this is a bit of an old thread, but recently we added support for Next.js on Azure Static Web Apps (see https://techcommunity.microsoft.com/t5/apps-on-azure-blog/extending-next-js-support-in-azure-static-web-apps/ba-p/3627975), which support things like SSR, API routes, and others.

Failing that, I'd be looking at hosting Next.js within a container on [Container Apps](https://learn.microsoft.com/azure/container-apps/overview?WT.mc_id=javascript-0000-aapowell), as the container would give you the simplest deployment model.
